src/testing/oem-debuggers.md

```diff
@@ -1,371 +1,493 @@
-  java > io.flutter plugins</strong></nobr>.
-  Double click **GeneratedProjectRegistrant** to open the
-  Java code in the edit pane.
+1. Choose the process to which you want to attach.
+   In following screenshot, the `com.example.my_app` process
+   using the **Emulator Pixel_5_API_33** was selected.
 
-Both the Dart and OEM debuggers are interacting with the same process.
-User either, or both, to set breakpoints, examine stack, resume execution...
-In other words, debug!
+   !['Flutter app in Android device displaying two buttons and a link.'](/assets/images/docs/testing/debugging/oem/choose-process-dialog.png){:width="50%"}
+   <div markdown="1">
+   Flutter app in Android device displaying two buttons and a link.
+   </div>
 
-  <img src='/assets/images/docs/testing/debugging/oem/dart-debugger.png' alt='screenshot of Android Studio in the Dart debug pane.'>{:width="100%"}
-  <br><center>The Dart debug pane with two breakpoints set in `lib/main.dart`</center>
+1. Locate the tab for **Android Debugger** in the **Debug** pane.
 
-  <img src='/assets/images/docs/testing/debugging/oem/android-debugger.png' alt='screenshot of Android Studio in the Android debug pane.'>{:width="100%"}
-  <br><center>The Android debug pane with one breakpoint set in
-   `GeneratedPluginRegistrant.java`.
-  Toggle between the debuggers by clicking the appropriate debugger in
-   the Debug pane's banner.</center>
+1. In the **Project** pane, expand
+   **_app_name__android > android > app > src > main > java > io.flutter plugins**.
 
-## Debugging with Xcode (iOS)
+1. Double click **GeneratedProjectRegistrant** to open the
+   Java code in the **Edit** pane.
+   
+   !['The Android Project view highlighting the GeneratedPluginRegistrant.java file.'](/assets/images/docs/testing/debugging/oem/debug-open-java-code.png){:width="100%"}
+   <div markdown="1">
+   The Android Project view highlighting the `GeneratedPluginRegistrant.java` file.
+   </div>
 
-In order to debug OEM iOS code, you need an app that contains OEM iOS code.
-In this section, you'll learn how to connect two debuggers to your app:
-1) the Dart debugger and, 2) the Xcode debugger.
+At the end of this procedure, both the Dart and Android debuggers interact
+with the same process.
+Use either, or both, to set breakpoints, examine stack, resume execution
+and the like. In other words, debug!
+
+![The Dart debug pane with two breakpoints set in `lib/main.dart`](/assets/images/docs/testing/debugging/oem/dart-debugger.png){:width="100%"}
+<div markdown="1">
+The Dart debug pane with two breakpoints set in `lib/main.dart`.
+</div>
+
+!['The Android debug pane with one breakpoint set in GeneratedPluginRegistrant.java.'](/assets/images/docs/testing/debugging/oem/android-debugger.png)
+<div markdown="1">
+The Android debug pane with one breakpoint set in GeneratedPluginRegistrant.java.
+</div>
+
+[`url_launcher`]: {{site.url}}/examples/testing/oem_debugging
+
+### Debug Dart and iOS code using Xcode
```

Comment:
Is there a reason you're not covering macOS here as well (which is almost identical), given that you are covering Windows?

Reply:
Only that there was no issue for it. #3847 was for iOS, #7600 was for Windows. I have no objection to adding the macOS code, other than there was no request to do so.

Reply:
Maybe out of scope here then. But adding it in a follow-up should be pretty trivial since other than some paths it should be essentially the same as iOS.